### PR TITLE
feat(map): Extend supported Key Types for C# codegen

### PR DESCRIPTION
**Pull request by jonathan-casey**
### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

### CODE REVIEW

File: test/codegen/__snapshots__/codegen.js.snap [hunk 1]
```diff
@@ -5980,7 +5980,7 @@ public abstract class Person : Participant {
    public string ssn { get; set; }
    public float height { get; set; }
    public System.DateTime dob { get; set; }
-   public Dictionary<string, KinTelephone> nextOfKin { get; set; }
+   public Dictionary<string, System.DateTime> nextOfKin { get; set; }
 }
```
Comment (mttrbrts) [inline test/codegen/__snapshots__/codegen.js.snap:5984]: These types seem to have got more general. Is this expected?

Comment (Jonathan-Casey) [reply test/codegen/__snapshots__/codegen.js.snap:5984]: Yes, the underlying Primitive of a Scalar should be printed, in place of its alias name

Comment (dselman) [reply test/codegen/__snapshots__/codegen.js.snap:5982]: Beware the logic that handles UUID. Did you test with a map that has a UUID as a key or value? It looks like the code generator doesn't unbox those.

Comment (jonathan-casey) [reply test/codegen/__snapshots__/codegen.js.snap:5982]: @dselman logic added to handle `concerto.scalar.UUID` for k & v

**Review by dselman — APPROVED** [review]

**Pull request merged**